plugin.video.eso/resources/lib/eso_addon.py

```diff
@@ -0,0 +1,228 @@
+# -*- coding: utf-8 -*-
+
+'''
+   KODI ESO video add-on.
+   Copyright (C) 2013 José Antonio Montes (jamontes)
+
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+   This is the first trial of the ESO video add-on for KODI.
+   This add-on gets the videos from ESO web site and shows them properly ordered.
+   You can choose the preferred subtitled language for the videos, if it is available.
+   This plugin depends on the lutil library functions.
+'''
+
+import resources.lib.lutil as lutil
+
+pluginhandle = lutil.get_plugin_handle()
+plugin_id = 'plugin.video.eso'
+
+settings = lutil.get_plugin_settings(plugin_id)
+lutil.set_debug_mode(settings.getSetting("debug"))
+translation = settings.getLocalizedString
+root_dir = settings.getAddonInfo('path')
+lutil.set_fanart_file(root_dir)
+st_release = settings.getSetting('version')
+current_release = settings.getAddonInfo('version')
+sort_url_param = '' if settings.getSetting("rated") == "true" else '?sort=-release_date'
+update_settings = False
+
+# This is to make it sure that settings are correctly setup on every addon update or on first run.
+if not st_release:
+    lutil.log("eso Warning: First run. Update settings.")
+    settings.openSettings()
+    settings.setSetting('version', current_release)
+elif st_release != current_release:
+    lutil.log("eso Warning: updated release. Check for update settings.")
+    if update_settings:
+        settings.openSettings()
+    settings.setSetting('version', current_release)
+
+# Gets the quality for videos from settings
+try:
+    quality = int(settings.getSetting('quality'))
```

Comment:
Suggestion: why not use the new `AddonSettings.setInt(id, value)`. Same goes for the other settings?.

See: https://xbmc.github.io/docs.kodi.tv/master/kodi-dev-kit/group__python__settings.html#gaa3a37ffb19146a4cc02da1898fc215e2